aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeDefinition;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.BillingMode;
+import software.amazon.awssdk.services.dynamodb.model.ConditionalCheckFailedException;
+import software.amazon.awssdk.services.dynamodb.model.CreateTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.DeleteItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableResponse;
+import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement;
+import software.amazon.awssdk.services.dynamodb.model.KeyType;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.dynamodb.model.ScalarAttributeType;
+import software.amazon.awssdk.services.dynamodb.model.TableStatus;
+
+/**
+ * DynamoDB implementation for the lock manager.
+ */
+class DynamoLockManager implements LockManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DynamoLockManager.class);
+
+  private static final String LOCK_TABLE_COL_LOCK_ID = "lockId";
+  private static final String LOCK_TABLE_COL_EXPIRE_TS_MILLIS = "expireTimestampMillis";
+  private static final long LOCK_TABLE_CREATION_WAIT_MS = 5000;
+  private static final int LOCK_TABLE_CREATION_WAIT_ROUNDS = 10;
+
+  private static final List<KeySchemaElement> LOCK_TABLE_SCHEMA = Lists.newArrayList(
+      KeySchemaElement.builder()
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .keyType(KeyType.HASH)
+          .build()
+  );
+
+  private static final List<AttributeDefinition> LOCK_TABLE_COL_DEFINITIONS = Lists.newArrayList(
+      AttributeDefinition.builder()
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .attributeType(ScalarAttributeType.S)
+          .build()
+  );
+
+  private final DynamoDbClient dynamo;
+  private final AwsProperties awsProperties;
+
+  DynamoLockManager(DynamoDbClient dynamo, AwsProperties awsProperties) {
+    this.dynamo = dynamo;
+    this.awsProperties = awsProperties;
+    ensureLockTableExistsOrCreate();
+  }
+
+  private void ensureLockTableExistsOrCreate() {
+    ensureTableExistsOrCreate(awsProperties.glueCatalogLockTable(), LOCK_TABLE_SCHEMA, LOCK_TABLE_COL_DEFINITIONS);
+  }
+
+  private void ensureTableExistsOrCreate(String tableName, List<KeySchemaElement> schema,
+                                         List<AttributeDefinition> definitions) {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
```

Comment:
This is duplicated below when waiting for the table to become active. I think it would be cleaner to have a `tableExists` method:

```java
  private boolean tableExists(String tableName) {
    try {
      dynamo.describeTable(...);
      return true;
    } catch (ResourceNotFoundException e) {
      return false;
    }
  }
```

Then you could call that method here as well as in `waitUntilActive`. That would avoid having so much logic in a `catch` block.